### DISCUSSION TITLE
How to configure cgroup related configuration in build

### DISCUSSION THREAD

**Discussion by Hariom-avin** [OE4T/meta-tegra#1083] [General]
Hi,

I want to configure cgroup related configuration in Yocto build. Please guide me how can I configure cgroup during build time in Yocto build.

Regards,
Hariom

**Comment by madisongh** [comment]
I've moved this to a discussion thread.

Can you be a bit more specific about what you're looking for?  The default kernel configuration for the Jetsons from NVIDIA already enables several control groups (cgroups) features to support things like docker.

**Reply by Hariom-avin** [reply]
Hi Mr. Madisongh,

I am working on yocto env to run qemu emulator.
As a parent process I have to restrict each process usage of cpu limit and memory limit. So I am using cgroup to limit the cpu and memory.
To limit memory I have used memory as controller and memory.limit_in_bytes as sub-controller.
To limit cpu I have used cpu as controller and cpu.cfs_quota_us as sub-controller.

When parent process try to create cgroup using cgroup_create_cgroup()  getting error as "Error: failed to set /sys/fs/cgroup/cpu/ResourceGroup1/cpu.cfs_quota_us: Cgroup, requested group parameter does not exist".

I am able to set limit for memory but not able to set cpu limit because parent process not able to find the sub-controller(cpu.cfs_quota_us)

That is why I want to configure cgroup related configuration in Yocto build.

**Reply by madisongh** [reply]
This repository is for integration of NVIDIA's Jetson Linux into Yocto builds. General questions about kernel configuration for other platforms would be better directed to other forums.  

Have you read the [kernel documentation on the scheduler](https://docs.kernel.org/scheduler/index.html)?  Have you added the necessary configuration options to your kernel build, as described in the [Yocto Project documentation](https://docs.yoctoproject.org/4.0.3/kernel-dev/common.html#configuring-the-kernel)?

**Reply by Hariom-avin** [reply]
Hi Mr. Madisongh,

I understand that this repository is for integration of NVIDIA's Jetson Linux into Yocto builds.

I will check the links shared by you. And I will raise the query to Yocto forum. 

Thanks for your guidance.

Regards,
Hariom
